packages/hardhat-core/src/builtin-tasks/compile.ts

```diff
@@ -724,16 +728,17 @@ export default function () {
         }: {
           compilationJob: CompilationJob;
           input: CompilerInput;
-          output: any;
+          output: CompilerOutput;
         },
         { artifacts, config, run }
       ): Promise<{
         artifactsEmittedPerFile: ArtifactsEmittedPerFile;
       }> => {
         const pathToBuildInfo = await artifacts.saveBuildInfo(
+          compilationJob.getSolcConfig().version,
+          "long version", // TODO-HH: Get the long version
```

Comment:
This is the only thing missing from this PR, @fvictorio.

I'm not sure how to do it. The `CompilerDownloader` has a method `getCompilerBuild` which can be used for this, but adding it here would break any compiler selection override.

I think we need to change the subtasks a little, but I'm not convinced about how.

Reply:
I think `TASK_COMPILE_SOLIDITY_GET_SOLC_PATH` could return the `solcLongVersion`, along with the `compielrPath` and the `isSolcJs` flag. (Maybe this could be an interface, I don't know).

Then `TASK_COMPILE_SOLIDITY_COMPILE_SOLC` could return both the output and information about the compiler used (we only care about the `solcLongVersion`, but maybe it's worth thinking about this in those terms).

This means that `TASK_COMPILE_SOLIDITY_COMPILE_JOB` receives both the output and the `solcLongVersion` and can pass both of them to `TASK_COMPILE_SOLIDITY_EMIT_ARTIFACTS`.

Reply:
I like this suggestion. I implemented it.